src/main/java/org/openrewrite/kotlin/KotlinParser.java

```diff
@@ -201,7 +206,8 @@ public Stream<SourceFile> parseInputs(Iterable<Input> sources, @Nullable Path re
                                     return (SourceFile) null;
                                 })
                                 .limit(1))
-                .filter(Objects::nonNull);
+                .filter(Objects::nonNull)
+                .filter(source -> !source.getSourcePath().getFileName().toString().matches("dependsOn-[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\\.kt"));
```

Comment:
What made you pick a generated file name? Might be nice to [similarly pick a reasonable filename](https://github.com/openrewrite/rewrite/blob/9dd7034d5b2479421ac0ccb57051d8886fac4c38/rewrite-java/src/main/java/org/openrewrite/java/JavaParser.java#L428-L449) where we can here as well, as compared to Java.

Reply:
I've copied the logic from JavaParser and polished it a bit + some small changes to make it Kotlin compatible